### DISCUSSION TITLE
Easier support for different user themes?

### DISCUSSION THREAD

**Discussion by steve6375** [a1ive/grub2-filemanager#305] [Ideas]
With the new Info F5 menus, it is becoming quite difficult to change the theme and make the four theme .txt files.
For instance, if a large font is used in a new theme, then the new F5 menu goes wrong if we keep the original info.txt theme file.
![image](https://user-images.githubusercontent.com/1469350/103168134-87d97800-4828-11eb-8870-67f677fdeb23.png)

It would be easier for the user if they only had to use their own theme.txt and fm.txt ($theme_std and $theme_fm) but not set a theme file for info.txt and help.txt - just use the default internal theme and use only the default font for those menus. Is this possible? Could the font be defined in the internal default .txt theme files? Could a more 'generic' theme be used for the info.txt and help.txt so it works with any theme and font the user may use?

**Comment by steve6375** [comment]
P.S. No hotkey for MOTHERBOARD ?
![image](https://user-images.githubusercontent.com/1469350/103168817-86ab4980-482e-11eb-835c-f36e105229dc.png)

**Comment by a1ive** [comment]
> P.S. No hotkey for MOTHERBOARD ?
> ![image](https://user-images.githubusercontent.com/1469350/103168817-86ab4980-482e-11eb-835c-f36e105229dc.png)

Oh, I forgot to add the hotkey.

**Comment by a1ive** [comment]
> For instance, if a large font is used in a new theme, then the new F5 menu goes wrong if we keep the original info.txt theme file.

Fonts cannot be scaled like images. Any non-16px fonts are not suitable for use with the built-in theme.

> Could a more 'generic' theme be used for the info.txt and help.txt so it works with any theme and font the user may use?

I don't think this can be done.
Even a complete operating system like iOS has display problems caused by different screen sizes (e.g. iPhone and iPad).

**Comment by steve6375** [comment]
If you add a font line into each of your .txt files, will your theme then use the correct font?

item_font = "Unifont Regular 16"

**Comment by steve6375** [comment]
```
+ hbox {
    top = 100%-80
    left = 25%
    width = 60%
    + vbox {
        + label {text = "[F1]" color = "#ffffff"  font = "Unifont Regular 16"}
        + image {file = "dock/help.png"}
    }
    + image {file = "dock/space.png"}
    + vbox {
        + label {text = "[F2]" color = "#ffffff" font = "Unifont Regular 16"}
        + image {file = "dock/fm.png"}
    }
    + image {file = "dock/space.png"}
    + vbox {
        + label {text = "[F3]" color = "#ffffff" font = "Unifont Regular 16"}
        + image {file = "dock/boot.png"}
    }
    + image {file = "dock/space.png"}
    + vbox {
        + label {text = "[F4]" color = "#ffffff" font = "Unifont Regular 16"}
        + image {file = "dock/settings.png"}
    }
    + image {file = "dock/space.png"}
    + vbox {
        + label {text = "[F5]" color = "#ffffff" font = "Unifont Regular 16"}
        + image {file = "dock/net.png"}
    }
    + image {file = "dock/space.png"}
    + vbox {
        + label {text = "[F6]" color = "#ffffff" font = "Unifont Regular 16"}
        + image {file = "dock/logout.png"}
    }
}
```
and also set font for F1-F6 icons ?

**Comment by steve6375** [comment]
and use
`  font = "Unifont Regular 16"`
for progress bar object?

**Reply by a1ive** [reply]
The default font of grub is set to "Unknown Regular 16" and the default terminal font is set to "Fixed10", which are not valid.
Set them to "Unifont Regular 16".
https://github.com/a1ive/grub/commit/803fa2d4725a08f93f78badf881b5b49a7555a38

**Comment by a1ive** [comment]
Now users could load external themes.
https://github.com/a1ive/grub2-filemanager/commit/3fe16e9b1950196f9ad70241301e4eeb48745849
```
export theme_hw_grub=${prefix}/themes/slack/hwinfo/grub.txt;
export theme_hw_cpu=${prefix}/themes/slack/hwinfo/cpu.txt;
export theme_hw_ram=${prefix}/themes/slack/hwinfo/ram.txt;
export theme_hw_board=${prefix}/themes/slack/hwinfo/board.txt;
```
